core/src/main/java/org/apache/iceberg/actions/BinPackStrategy.java

```diff
@@ -46,6 +46,7 @@
  * RewriteDataFiles#TARGET_FILE_SIZE_BYTES}.
  *
  * @deprecated since 1.3.0, will be removed in 1.4.0; use {@link SizeBasedFileRewriter} instead.
+ *     Note: This can only be removed once Spark 3.1 + 3.2 isn't using this API anymore.
```

Comment:
Maybe we should update the Spark 3.1/3.2 usage in some other PRs.

Reply:
yes that would be great, but I'm not sure how feasible this is, given that we don't backport every feature to old Spark versions

Reply:
I think that would be a non-trivial process, which is probably not worth it.